### DISCUSSION TITLE
Will there be a plugin update for UE 4.27 ?

### DISCUSSION THREAD

**Discussion by ghost** [Sixze/ALS-Refactored#25] [General]
Are you planning to upgrade the plugin to UE 4.27 ?

**Comment by Sixze** [comment]
It is compatible with both 4.27 and 5.0 EA, just change `EngineVersion` inside `ALS.uplugin`.

**Reply by ghost** [reply]
I tested in 4.27 and 5.0EA2

In 4.27 it compiles correctly.
Packaging the project works correctly.

However, in UE5 there is a problem with ragdoll, the character does a flip when falling.
Packing the project works correctly.

There is also a slight problem with falling, you can see it in 4.26, 4.27 and 5.0.
There is a jerk at the end of the fall.

**Reply by Sixze** [reply]
The ragdoll issues in 5.0 EA are related to the Chaos physics engine, not ALS.

Can you tell me more about jerk at the end of the fall? Is it related to ragdoll or occurs during normal fall? I would be grateful for a short video with this issue.

**Reply by ghost** [reply]
I am attaching the video.
https://streamable.com/q5b8n0

Earlier I packaged the project in UE 5 but did not test it, now I run it and got the message that it can not load the map "playground", so I created in the folder "Content" folder Maps and there I overwrote this map, I set it again in the project settings, I did "build" I saved and compiled, after running it no longer had the message that it can not load the map, it just loaded it incorrectly which you can see in the picture, during compilation there were more than 700 warnings.

![ue5_ALS_after_packing_bug](https://user-images.githubusercontent.com/11320046/135757756-4f7991e6-fabe-486f-b2a0-6c6902dab1a3.png)

and only a part of the map showed up untextured and without characters.
[warnings.txt](https://github.com/Sixze/ALS-Refactored/files/7273461/warnings.txt)

**Reply by Sixze** [reply]
I see the problem with jerking and will try to figure it out in the future. It looks like the original ALS and ALS-Community also have the same problem.

Are you sure you added the map from the plugin to the list of maps for packaging, as described here: https://www.reddit.com/r/unrealengine/comments/6dq3ke/tip_if_your_packaging_games_with_multiple_maps/?

**Reply by ghost** [reply]
I will follow the project on a regular basis :-)
I still noticed a bug when the character is on a large hill in a crouched position, during rotation the leg behaves in a very unnatural way, as you need I will send a video how it looks like.

Back to the map, I have now included this map in the packing settings, restarted the engine, when packing there are the same errors and the map looks like above.

Perhaps this is simply a bug of an early version of the engine.
![ue5](https://user-images.githubusercontent.com/11320046/135762326-50355c90-d12a-4077-bffd-db15a98c46c5.png)

**Reply by Sixze** [reply]
I checked on an empty project, in 5.0 EA packaged build works. You just need to remove the line with `EngineVersion` in `ALS.uplugin` and in the project settings set `GameDefaultMap` to the map from the plugin.

**Reply by Sixze** [reply]
It looks like the problem with jerking during landing is a graphical bug caused by motion blur when a moving object stops instantly. Motion blur, by default, is also influenced by the time scale, so it basically doesn't work very well in slow motion. Camera movement and character animations also affect the perception of this problem.

**Reply by ghost** [reply]
I removed the line you pointed out and it worked :-)
Previously a window popped up saying the plugin was incompatible and asking if I wanted to continue, I thought if I selected "yes" it was equivalent to skipping that line.
Anyway sorry for the confusion :-)

If this is about the micro jerk then thanks for the explanation :-)
